mingw-w64 compilation failing, rpl_malloc undefined

seems like some defines not expected in the Windows toolchain are being used

```
make[2]: Entering directory '/home/chungy/Games/doom/woof/Source'
i686-w64-mingw32-gcc -DHAVE_CONFIG_H -I. -I..    -I/usr/i686-w64-mingw32/include/SDL2 -Dmain=SDL_main  -I/usr/i686-w64-mingw32/include/SDL2 -Dmain=SDL_main  -I/usr/i686-w64-mingw32/include/SDL2 -Dmain=SDL_main  -g -O2 -g -O2 -Wall -Wdeclaration-after-statement -Wredundant-decls -MT woof-z_zone.o -MD -MP -MF .deps/woof-z_zone.Tpo -c -o woof-z_zone.o `test -f 'z_zone.c' || echo './'`z_zone.c
In file included from doomdef.h:34,
                 from d_items.h:32,
                 from d_player.h:36,
                 from d_net.h:32,
                 from doomstat.h:40,
                 from z_zone.c:38:
../config.h:178: warning: "malloc" redefined
  178 | #define malloc rpl_malloc
      | 
In file included from z_zone.c:37:
z_zone.h:91: note: this is the location of the previous definition
   91 | #define malloc(n)          Z_Malloc(n,PU_STATIC,0)
      | 
In file included from doomdef.h:34,
                 from d_items.h:32,
                 from d_player.h:36,
                 from d_net.h:32,
                 from doomstat.h:40,
                 from z_zone.c:38:
../config.h:181: warning: "realloc" redefined
  181 | #define realloc rpl_realloc
      | 
In file included from z_zone.c:37:
z_zone.h:93: note: this is the location of the previous definition
   93 | #define realloc(p,n)       Z_Realloc(p,n,PU_STATIC,0)
      | 
In file included from info.h:35,
                 from p_pspr.h:46,
                 from d_player.h:37,
                 from d_net.h:32,
                 from doomstat.h:40,
                 from z_zone.c:38:
z_zone.c: In function ‘Z_Init’:
../config.h:178:16: error: ‘rpl_malloc’ undeclared (first use in this function); did you mean ‘_mm_malloc’?
  178 | #define malloc rpl_malloc
      |                ^~~~~~~~~~
z_zone.c:236:22: note: in expansion of macro ‘malloc’
  236 |   while (!(zonebase=(malloc)(zonebase_size=size + HEADER_SIZE + CACHE_ALIGN)))
      |                      ^~~~~~
../config.h:178:16: note: each undeclared identifier is reported only once for each function it appears in
  178 | #define malloc rpl_malloc
      |                ^~~~~~~~~~
z_zone.c:236:22: note: in expansion of macro ‘malloc’
  236 |   while (!(zonebase=(malloc)(zonebase_size=size + HEADER_SIZE + CACHE_ALIGN)))
      |                      ^~~~~~
z_zone.c: In function ‘Z_Malloc’:
../config.h:178:16: error: ‘rpl_malloc’ undeclared (first use in this function); did you mean ‘_mm_malloc’?
  178 | #define malloc rpl_malloc
      |                ^~~~~~~~~~
z_zone.c:387:21: note: in expansion of macro ‘malloc’
  387 |    while(!(block = (malloc)(size + HEADER_SIZE)))
      |                     ^~~~~~
make[2]: *** [Makefile:1584: woof-z_zone.o] Error 1
make[2]: Leaving directory '/home/chungy/Games/doom/woof/Source'
```

--- a/configure.ac
+++ b/configure.ac
@@ -54,8 +54,6 @@ AC_CHECK_TYPES([ptrdiff_t])
 # Checks for library functions.
 AC_FUNC_ALLOCA
 AC_FUNC_ERROR_AT_LINE
-AC_FUNC_MALLOC
-AC_FUNC_REALLOC
 AC_CHECK_FUNCS([atexit memmove memset mkdir pow putenv strcasecmp strchr strdup strerror strncasecmp strrchr strstr strtol])
 
 AC_ARG_ENABLE([dogs], AS_HELP_STRING([--disable-dogs], [Disable support for helper dogs]))

